File: atividade-9/atividade.py
from abc import ABC, abstractmethod

class Forma(ABC):
    @abstractmethod
    def area(self):
        pass

class Quadrado(Forma):
    def __init__(self, lado):
        self.lado = lado

    def area(self):
        return self.lado ** 2

class Circulo(Forma):
    def __init__(self, raio):
        self.raio = raio

    def area(self):
        import math
        return math.pi * self.raio ** 2

quadrado = Quadrado(4)
circulo = Circulo(3)

print(f"Área do quadrado: {quadrado.area()}")
print(f"Área do círculo: {circulo.area()}")

File: atividade-9/atividade.go
package main

import (
	"fmt"
	"math"
)

type Forma interface {
	Area() float64
}

type Quadrado struct {
	Lado float64
}

func (q Quadrado) Area() float64 {
	return q.Lado * q.Lado
}

type Circulo struct {
	Raio float64
}

func (c Circulo) Area() float64 {
	return math.Pi * c.Raio * c.Raio
}

func main() {
	quadrado := Quadrado{Lado: 4}
	circulo := Circulo{Raio: 3}

	fmt.Printf("Área do quadrado: %.2f\n", quadrado.Area())
	fmt.Printf("Área do círculo: %.2f\n", circulo.Area())
}
